### DISCUSSION TITLE
Checking if database exists

### DISCUSSION THREAD

**Discussion by ericp-mrel** [wintercms/winter#547] [Q&A] [answered]
Is there an efficient way of checking whether or not the database exists during the plugin boot process to prevent errors when installing plugins on a fresh Winter CMS project which doesn't have a database yet?

I have a plugin which does some checks on the plugin's settings model, but throws errors when it boots the plugin during the composer install process.

Thanks

**Comment by bennothommo** [comment, marked as answer]
@ericp-mrel there is indeed.

```
$databaseExists = App::hasDatabase();
```

**Reply by ericp-mrel** [reply]
Awesome, Thanks!
